test/truffle/compiler/graphs/run

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env ruby
+
+NATIVE = `tool/jt.rb ruby --version` =~ /Native/
+
+$tests = 0
+$failures = 0
+
+def test(program, extra='', jvm_nodes_expected, native_nodes_expected, shape_expected)
+  nodes_expected = NATIVE ? native_nodes_expected : jvm_nodes_expected
+  $tests += 1
+  got = `tool/jt.rb graph --describe #{program} #{extra} 2>&1`.lines.last.strip
+  unless got =~ /(\d+) nodes, (.*)/
+    puts "#{program}: error #{got.inspect}"
+    $failures += 1
+  end
+  nodes_got = $1.to_i
+  shape_got = $2
+  # We could be fuzzy about the number of nodes if we wanted to
+  if nodes_got == nodes_expected && shape_got == shape_expected
+    puts "#{program}: ✓"
+  else
+    puts "#{program}: expected #{nodes_expected} #{shape_expected}, got #{nodes_got} #{shape_got}"
+    $failures += 1
+  end
+end
+
+test 'test/truffle/compiler/graphs/args-req.rb', 36, 34, 'linear'
+test 'test/truffle/compiler/graphs/args-opt-set.rb', 36, 34, 'linear'
+test 'test/truffle/compiler/graphs/args-opt-unset.rb', 32, 31, 'linear'
+test 'test/truffle/compiler/graphs/args-rest.rb', 36, 34, 'linear'
```

Comment:
I don't think we want to hardcode node numbers here, as explained in my other comment, because it's too brittle.
The fact it differs between native and not is already a bad sign for stability.